examples/swash_render/src/main.rs

```diff
@@ -0,0 +1,223 @@
+    let run_y = glyph_run.baseline();
+    let style = glyph_run.style();
+    let color = style.brush;
+
+    // Get the "Run" from the "GlyphRun"
+    let run = glyph_run.run();
+
+    // Resolve properties of the Run
+    let font = run.font();
+    let font_size = run.font_size();
+    let normalized_coords = run.normalized_coords();
+
+    // Convert from parley::Font to swash::FontRef
+    let font_ref = FontRef::from_index(font.data.as_ref(), font.index as usize).unwrap();
+
+    // Iterates over the glyphs in the GlyphRun
+    for glyph in glyph_run.glyphs() {
+        let glyph_id: GlyphId = glyph.id;
+        let glyph_x = run_x + glyph.x;
+        let glyph_y = run_y - glyph.y;
+        run_x += glyph.advance;
+        let Some(rendered_glyph) = render_glyph(
+            context,
+            &font_ref,
+            font_size,
+            normalized_coords,
+            glyph_id,
+            glyph_x.fract(),
+            glyph_y.fract(),
+        ) else {
+            println!("No glyph");
+            continue;
+        };
+
+        let glyph_width = usize::try_from(rendered_glyph.placement.width).expect("usize < 32 bits");
+        let glyph_height =
+            usize::try_from(rendered_glyph.placement.height).expect("usize < 32 bits");
+        let glyph_origin_x =
+            glyph_x.floor() as i32 + rendered_glyph.placement.left + padding as i32;
+        let glyph_origin_y =
+            (glyph_y.floor() as i32) - rendered_glyph.placement.top + padding as i32;
+
+        match rendered_glyph.content {
+            Content::Mask => {
+                let mut i = 0;
+                for off_y in 0..glyph_height as i32 {
+                    for off_x in 0..glyph_width as i32 {
+                        let x = (glyph_origin_x + off_x) as u32;
+                        let y = (glyph_origin_y + off_y) as u32;
+                        let alpha = rendered_glyph.data[i];
+                        let color = Rgba([color.r, color.g, color.b, alpha]);
+                        img.get_pixel_mut(x, y).blend(&color);
+                        i += 1;
+                    }
+                }
+            }
+            Content::SubpixelMask => unimplemented!(),
+            Content::Color => {
+                for (off_y, row) in rendered_glyph
+                    .data
+                    .chunks_exact(glyph_width * 4)
+                    .enumerate()
+                {
+                    for (off_x, pixel) in row.chunks_exact(4).enumerate() {
+                        let &[r, g, b, a] = pixel else {
+                            panic!("Pixel doesn't have 4 components")
+                        };
+                        let x = (glyph_origin_x + off_x as i32) as u32;
+                        let y = (glyph_origin_y + off_y as i32) as u32;
+                        let color = Rgba([r, g, b, a]);
+                        img.get_pixel_mut(x, y).blend(&color);
+                    }
+                }
+            }
+        };
```

Comment:
@dfrg Would you be open to adding this pixel iteration code into swash itself(I'm thinking as a method on `Mask` or `swash::Image`)? It would be a nice simplification for this example and could also be used in cosmic-text which has some similar iteration code.

Reply:
(I don't think we should block this PR on that)

Reply:
Yes, that seems like a useful addition.